src/components/keyboard-nav.js

```diff
@@ -0,0 +1,92 @@
+import _ from 'lodash/fp'
+import { useRef } from 'react'
+import * as Utils from 'src/libs/utils'
+
+
+export const withArrowKeyNavigation = ({ onKeyDown, children }) => {
+  const count = children.length
+  const refs = _.map(useRef, _.range(0, count))
+
+  const focusOn = index => {
+    // Wrap around the ends and ensure the number is positive
+    const i = index % count + (index < 0 ? count : 0)
+    refs[i].current?.focus()
+  }
+
+  // Pass a ref into each child, using an alternative property name
+  return _.map(([i, child]) => {
+    return child({ forwardedRef: refs[i], onKeyDown: onKeyDown(i, focusOn) })
```

Comment:
Maybe `cloneElement`?

Reply:
This might be moot if I end up rewriting the ref passing. But... why add a clone here? This should be the only place the element is used, right? How would returning a clone rather than the original affect the ref?